docs/decisions/0001-using-helm-hook-with-kubectl-pod-for-configuration-changes.md

```diff
@@ -0,0 +1,138 @@
+# Using Helm-Hooks with Kubectl Pod for Configuration Changes
+
+status: proposed \
+date: 2022-06-03 \
+deciders: \
+consulted: \
+informed:
+
+## Context and Problem Statement
+
+In the face of improving our reliability, we need to apply some changes to applications, which are not yet in a HA setup.
+The HA setup for each application shall be pursued separately and supersede this proposal.
+
+How to configure running applications which provide an API for that? \
+How to avoid downtimes doing so? \
+How to current need handle database / user-changes? \
+How to minimise effort in face of upcoming transition to HA solutions? \
+How do we handle a rollback? \
+How do we avoid locking ourselves out?
+
+## Considered Options
+
+* Helm-Chart Hook
+    * Job with Same Image
+    * Job with Different Image
+* Wait for Proper Solution
+* Restart Only
+    * Checksum-annotations
+    * PostStart-Hook
+    * Reloader
+* Move it to Pipeline
+* Monitoring
+    * Same Container
+    * Same Pod, Different Container / Same Image
+    * Same Pod, Different Container / Different Image
+    * Different Pod / Same Image
+    * Operator (Different Pod / Other Image)
+
+## Decision Outcome
+
+Chosen option: "Helm-chart hook with different image", because it comes out best (see below).
+
+### Positive Consequences
+
+* We gain the ability to apply configuration changes without downtime
+* We only need to modify the helm-charts
+  * No changes to the pipelines,
+  * Upstream images can be used
+* Rollback can be easily handled by implementing the corresponding hook.
+
+### Negative Consequences
+
+* The service charts gain the dependency to another image, which needs to be updated.
+  However, since it is limited to deployment time, there is no need to trigger a deployment
+  on change.
+
+## Pros and Cons of the Options
+
+Almost all options exists as "Same Image" and "Different Image" variant, so first have a look at those.
+
+### Same Image
+
+One can use the same image as the one providing the application and add a layer
+which contains the logic of applying the configuration, or use a completely separate image.
+It is the only option, when wanting to run the logic in the same container as the application
+
+* Good, because it would be a single dependency
+* Good, because the configuration tools are included in the image already
+* Neutral, because it would be an application specific solution
+* Bad, because we need extra pipelines to add out configuration logic to the images
+* Bad, because a change of the configuration logic would/should always trigger a deployment, ie. downtime
+
+### Different Image
+
+The opposite of "Same Image".
+
+### Helm-Chart Hook
+
+We could make use of helm-chart hooks for upgrade and rollback.
```

Comment:
May be good to link to https://helm.sh/docs/topics/charts_hooks/ here. I was confused what was meant by this since I had not heard of this feature before.

Reply:
I've added the link